src/simsopt/field/magneticfield.py

```diff
@@ -94,25 +94,43 @@ def to_vtk(self, filename, nr=10, nphi=10, nz=10, rmin=1.0, rmax=2.0, zmin=-0.5,
 
     def to_mgrid(self, filename, nr=10, nphi=4, nz=12, rmin=1.0, rmax=2.0, zmin=-0.5, zmax=0.5, nfp=1,
                  include_potential=False):
-        """Export the field to the mgrid format for free boundary calculations.
-
+        """Export the field to the mgrid file format for free boundary calculations.
+
+        An mgrid file contains a grid in cylindric coordinates upon which the
+        magnetic field is evaluated. For free boundary calculations, the
+        grid should be large enough to contain the entire plasma. 
+        The grid is defined by the number of
+        points in the major radius (``nr``), the number of planes per field period 
+        in the toroidal angle (``nphi``), and the number of points in the z coordinate (``nz``). 
+        A good rule here is to choose at least 4 times as many toroidal planes as the maximum toroidal 
+        mode number (ex. if ``ntor=6`` then you should have at least 24 toroidal planes). 
+        
+        The grid boundary is defined by the minimum and maximum values of the major radius
+        (``rmin``, ``rmax``), the minimum and maximum values of the z-coordinate (``zmin``, ``zmax``).
+        The choice of the number or radial and vertical gridpoints is not as straightforward.
+        The VMEC code uses these grids to 'deform' the plasma boundary in a iterative sense.
+        The complication revolves around the spectral condensation VMEC preforms in the poloidal direction.
+        The code calculates the location of the poloidal grid points so that an optimized choice is made for 
+        the form of the plasma (more point where needed, points, and less where they're not, straight segments).
+        The user should decide how accurately the wish to know the plasma boundary.
+        If [cm] precision is required then the number of grid points chosen should provide at least this resolution.
+        Remember that the more datapoints, the slower VMEC will run.
+        
         The field data is represented as a single "current group". For
```

Comment:
what is a current group? what is meant by field data?

Reply:
About current groups: these are sets of coils that are electrically connected in series such that their currents always scale proportionally to each other. (BTW the coils in a given current group need not have the same number of Ampere-turns because they can have different numbers of turns.) Consider W7-X, where each unique coil shape is repeated 10 times. W7-X has 7 current groups, one for each base curve (5 nonplanar coils + 2 planar coils). In mgrid files and free-boundary vmec, you can scale the currents in each current group individually. For example you can turn on all the planar coils without changing the currents in the nonplanar coils. In free-boundary vmec, the currents in each current group are scaled using the "extcur" array in the input files.

Reply:
thanks @landreman, this seems like useful docstring info to include @mishapadidar